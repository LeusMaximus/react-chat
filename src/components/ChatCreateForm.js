// React
import React from 'react';

// MUI components
import { withStyles } from '@material-ui/core';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

// Own modules
import isTextFieldValid from '../utils/isTextFieldValid';

const styles = theme => ({
  button: {
    marginTop: theme.spacing.unit * 4,
  },
});

class ChatCreateForm extends React.Component {
  state = {
    chatName: {
      value: '',
      isValid: true,
    },
  };

  validate = () => {
    const { chatName } = this.state;
    const isChatNameValid = isTextFieldValid(chatName.value);

    this.setState({
      chatName: {
        ...chatName,
        isValid: isChatNameValid,
      },
    });

    return isChatNameValid;
  }

  handleChange = (event) => {
    const { name, value } = event.target;

    this.setState(prevState => ({
      [name]: {
        ...prevState[name],
        value,
      },
    }));
  }

  handleSubmit = (event) => {
    event.preventDefault();

    const isValid = this.validate();

    if (!isValid) return;

    const { chatName } = this.state;

    this.props.onSubmit(chatName.value);
  }

  render() {
    const { classes } = this.props;
    const { chatName } = this.state;

    return (
      <form noValidate autoComplete="off" onSubmit={this.handleSubmit}>
        <TextField
          label="Type chat name"
          name="chatName"
          value={chatName.value}
          fullWidth
          required
          autoFocus
          margin="normal"
          onChange={this.handleChange}
          error={!chatName.isValid}
        />

        <Button
          variant="raised"
          color="primary"
          fullWidth
          className={classes.button}
          type="submit"
        >
          Create
        </Button>
      </form>
    );
  }
}

export default withStyles(styles)(ChatCreateForm);
