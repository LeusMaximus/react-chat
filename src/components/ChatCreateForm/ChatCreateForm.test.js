/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatCreateForm from './index';

const mockProps = {
  onSubmit: jest.fn(),
};

describe('<ChatCreateForm />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatCreateForm {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot test', () => {
    const tree = renderer.create(<ChatCreateForm {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
