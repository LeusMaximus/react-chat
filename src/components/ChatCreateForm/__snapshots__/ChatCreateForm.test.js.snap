// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatCreateForm /> snapshot test 1`] = `
<form
  autoComplete="off"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-56 MuiFormControl-marginNormal-57 MuiFormControl-fullWidth-59"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-65 MuiInputLabel-root-60 MuiInputLabel-formControl-61 MuiInputLabel-animated-64"
      data-shrink={false}
      htmlFor={undefined}
    >
      Type chat name
      <span
        className="MuiFormLabel-asterisk-69"
      >
        â€‰*
      </span>
    </label>
    <div
      className="MuiInput-root-70 MuiInput-fullWidth-77 MuiInput-formControl-71 MuiInput-underline-74"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete={undefined}
        autoFocus={true}
        className="MuiInput-input-78"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name="chatName"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-99 MuiButton-root-83 MuiButton-raised-89 MuiButton-raisedPrimary-90 MuiButton-fullWidth-98 ChatCreateForm-button-55"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="submit"
  >
    <span
      className="MuiButton-label-84"
    >
      Create
    </span>
    <span
      className="MuiTouchRipple-root-102"
    />
  </button>
</form>
`;
